utils/common/src/main/kotlin/EnvironmentVariableFilter.kt

```diff
@@ -0,0 +1,117 @@
+ *
+ * So in order to determine whether a specific variable "E" can be passed to a child process, this filter applies the
+ * following steps:
+ * - If E is contained in the allow list, it is included.
+ * - Otherwise, E is included if and only if it is not matched by one of the exclusion patterns.
+ */
+object EnvironmentVariableFilter {
+    /**
+     * A set with patterns for variable names that are denied by default. All variables containing one of these
+     * strings (ignoring case) are not propagated to child processes.
+     */
+    val DEFAULT_DENY_PATTERNS = setOf(
+        "key",
+        "pass",
+        "pwd",
+        "token",
+        "user"
+    )
+
+    /** A set with known variable names that are accepted despite they are matched by deny patterns. */
+    val DEFAULT_ALLOW_NAMES = setOf(
+        "CARGO_HTTP_USER_AGENT",
+        "COMPOSER_ALLOW_SUPERUSER",
+        "CONAN_LOGIN_ENCRYPTION_KEY",
+        "CONAN_LOGIN_USERNAME",
+        "CONAN_PASSWORD",
+        "CONAN_USERNAME",
+        "CONAN_USER_HOME",
+        "CONAN_USER_HOME_SHORT",
+        "DOTNET_CLI_CONTEXT_ANSI_PASS_THRU",
+        "GIT_ASKPASS",
+        "GIT_HTTP_USER_AGENT",
+        "GRADLE_USER_HOME",
+        "HACKAGE_USERNAME",
+        "HACKAGE_PASSWORD",
+        "HACKAGE_KEY",
+        "PWD",
+        "USER",
+        "USERPROFILE"
+    )
+
+    /** Stores the current deny patterns used by this filter. */
+    private var denyPatterns = DEFAULT_DENY_PATTERNS.toPatterns()
+
+    /** Stores the current allow names used by this filter. */
+    private var allowNames = DEFAULT_ALLOW_NAMES
+
+    /**
+     * Reset this filter to use the given [denyPatterns] and [allowNames]. The [denyPatterns] are
+     * converted to regular expressions, so that they can be matched anywhere in variable names, ignoring case.
+     */
+    fun reset(
+        denyPatterns: Collection<String> = DEFAULT_DENY_PATTERNS,
+        allowNames: Collection<String> = DEFAULT_ALLOW_NAMES
+    ) {
+        this.denyPatterns = denyPatterns.toPatterns()
+        this.allowNames = allowNames.toSet()
+    }
+
+    /**
+     * Test whether the variable with the given [name] can be passed to a child process according to the criteria
+     * described in the header comment.
+     */
+    fun isAllowed(name: String): Boolean {
+        return name in allowNames || denyPatterns.none { it.matches(name) }
+    }
+
+    /**
+     * Remove all keys from [environment] that do not pass this filter.
+     */
+    fun filter(environment: MutableMap<String, String>): MutableMap<String, String> {
+        val keysToRemove = environment.keys.filterNot(EnvironmentVariableFilter::isAllowed)
+
+        environment -= keysToRemove.toSet()
+
+        return environment
+    }
+
+    /**
+     * Convert the strings in this collection to deny pattern regular expressions.
+     */
+    private fun Collection<String>.toPatterns() = map { Regex(".*$it.*", RegexOption.IGNORE_CASE) }
```

Comment:
This basically just implements a case-insensitive sub-string search. So why use a regex at after, instead of simply using `denyPatterns.none { it.contains (name, ignoreCase = true) }` above?

Reply:
You are right. This is by far easier and completely sufficient. Changed.